superseded by #1613 - fix(btn): improve focus style to meet WCAG 2.2 AA

dancormier opened the pull request:
> **Note**
> This PR has been superseded by #1613

Addresses [STACKS-537](https://stackoverflow.atlassian.net/browse/STACKS-537), partially addresses [A11Y-2](https://stackoverflow.atlassian.net/browse/A11Y-2).

---

This PR updates the focus styles for the button component to meet the WCAG AA requirements.

> **Note**
> This PR includes newly generated baseline images for the button component. AFAICT, these images are not significantly different from the preexisting baseline images, with the exception of a few now-fixed [_muted w/ badge_ permutations](https://github.com/StackExchange/Stacks/pull/1610/files#diff-94f2f31e18981d99b453ba21c7ca061a1490806ac3dfbf3e91c0207ad4689f78).

dancormier committed:
fix(btn): improve focus style to meet WCAG 2.2 AA

Addresses [STACKS-537](https://stackoverflow.atlassian.net/browse/STACKS-537).
Partially addresses [A11Y-2](https://stackoverflow.atlassian.net/browse/A11Y-2)

dancormier committed:
fix(btn): improve focus style to meet WCAG 2.2 AA

Addresses [STACKS-537](https://stackoverflow.atlassian.net/browse/STACKS-537).
Partially addresses [A11Y-2](https://stackoverflow.atlassian.net/browse/A11Y-2)

dancormier committed:
refactor(btn): clean up organization of custom properties

dancormier committed:
refactor(btn): clean up organization of custom properties

dancormier committed:
refactor(btn): remove commented our focus code

dancormier committed:
refactor(btn): remove commented our focus code

dancormier committed:
fix(btn): outlined hover style; radio btn focus styles

dancormier committed:
fix(btn): outlined hover style; radio btn focus styles

dancormier committed:
fix(btn): fine tune focus rings alignment

dancormier committed:
fix(btn): fine tune focus rings alignment

dancormier committed:
chore(btn): regenerate base visual regression images

dancormier committed:
chore(btn): regenerate base visual regression images

dancormier committed:
2.0.9

giamir reviewed (commented):
The PR looks good. The new focus behavior seems to work as expected to me. 
Great work! 🎉
 
The only ask I have is to first merge and rebase your work off [this PR](https://github.com/StackExchange/Stacks/pull/1611) that improves the way we update our baseline images and keep the number of snapshot needing to be updated here to 72 (24 x 3  browsers). Thank you 😊

giamir committed:
test(visual): add improved baseline update workflow (#1611)

dancormier committed:
Revert "chore(btn): regenerate base visual regression images"

This reverts commit ea9b358bbde080889d32765be287791fee46749b.

dancormier committed:
Revert "chore(btn): regenerate base visual regression images"

This reverts commit ea9b358bbde080889d32765be287791fee46749b.

dancormier committed:
Merge branch 'STACKS-537/focus-style-button' of https://github.com/StackExchange/Stacks into STACKS-537/focus-style-button

dancormier committed:
test(visual): update button baseline images

dancormier committed:
Revert "test(visual): update button baseline images"

This reverts commit 2412ebfbb89bbdfb439cb5e4668e68d3c314d5a8.

dancormier committed:
test(visual): update button baseline images (take 2… or 3?)

dancormier committed:
refactor(btn): add stacked box-shadow instead of outline

Box-shadow tends to align more consistently. A transparent outline is used to ensure that an outline doesn't render when expected but can still be rendered in forced-color modes (see https://blogs.windows.com/msedgedev/2020/09/17/styling-for-windows-high-contrast-with-new-standards-for-forced-colors/#gist105426765)

dancormier commented:
This PR has been superseded by #1613
